module_prototyper/models/module_prototyper.py

```diff
@@ -0,0 +1,456 @@
+#    This module copyright (C) 2010 - 2014 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+import os
+import re
+import base64
+from datetime import date
+
+from collections import namedtuple
+from jinja2 import Environment, FileSystemLoader
+from openerp import models, api, fields
+
+YEAR = date.today().year
+
+
+class ModulePrototyper(models.Model):
+    """Module Prototyper gathers different information from all over the
+    database to build a prototype of module.
+    We are calling it a prototype as it will most likely need to be reviewed
+    by a developer to fix glitch that would sneak it during the generation of
+    files but also to add not supported features.
+    """
+    _name = "module_prototyper"
+    _description = "Module Prototyper"
+
+    license = fields.Selection(
+        [('GPL-2', 'GPL Version 2'),
+         ('GPL-2 or any later version', 'GPL-2 or later version'),
+         ('GPL-3', 'GPL Version 3'),
+         ('GPL-3 or any later version', 'GPL-3 or later version'),
+         ('AGPL-3', 'Affero GPL-3'),
+         ('Other OSI approved licence', 'Other OSI Approved Licence'),
+         ('Other proprietary', 'Other Proprietary')],
+        string='License',
+        default='AGPL-3',
+    )
+    name = fields.Char(
+        'Technical Name', required=True,
+        help=('The technical name will be used to define the name of '
+              'the exported module, the name of the model.')
+    )
+    category_id = fields.Many2one('ir.module.category', 'Category')
+    human_name = fields.Char(
+        'Module Name', required=True,
+        help=('The Module Name will be used as the displayed name of the '
+              'exported module.')
+    )
+    summary = fields.Char('Summary', required=True,
+                          help=('Enter a summary of your module'))
+    description = fields.Text(
+        'Description',
+        required=True,
+        help=('Enter the description of your module, what it does, how to'
+              'install, configure and use it, the roadmap or known issues.'
+              'The description will be exported in README.rst'),
+        default="""
+Module name
```

Comment:
Hi, 
This text is very huge.
Maybe it could be cleaner to move the whole text into a separated file, using a default function to get it ?